### ISSUE TITLE
Proper treatment of linebreaks

### ISSUE THREAD

**Issue by jhrmnn** [textX/textX#58]
I'm dealing with a language where the end of line has a meaning (Fortran). Naturally, there is a mechanism for including a linebreak (using the `&` character). Currently, I deal with this by including `& *\n *` as a possible `Comment`. This works great, but I was wondering if this approach has some potential catches, or if there is a better way to do this. I feel that a linebreak is more like a whitespace, but whitespaces can be only single characters.

**Comment by igordejanovic** [comment]
Yes, handling of whitespaces/comments is probably something that I should have done better. :)

At the moment they are handled in a similar fashion. I planed to keep comments and let the user access it somehow at they might contain useful information. Arpeggio keeps them on the next parsed object but I still haven't published that part of the API for general consumption. Whitespaces are defined as single chars for speed only if I remember correctly. In the parglare project which started later I followed different route by treating both whitespaces and comments as something called layout, i.e. a part of text that carry no semantics but might be useful to the reader and thus layout is fully accessible and consists of both whitespaces and comments and is defined as a separate grammar (like Comment rule in textX).

So, the bottom line is that your solution is right in the current setup. I don't have plans to rework that part in a backward incompatible way in 1.x series.

I have registered a poll in PR #54 to see the opinion on the `skipws` so that might impact you as if this change is introduced you will have to handle both whitespaces and comments in the grammar in all `noskipws` rules. This change would be backward incompatible so I'm still thinking if that would be a good idea.
